manual/src/cmds/intf-c.etex

```diff
@@ -1221,21 +1221,28 @@ has taken place since "r" was allocated.
 \subsection{ss:c-process-pending-actions}{Pending actions and asynchronous exceptions}
 
 Since 4.10, allocation functions are guaranteed not to call any OCaml
-callbacks from C, including finalisers and signal handlers, and delay
-their execution instead.
+callbacks from C, including finalisers and signal handlers, and to
+delay their execution instead.
 
 The function \verb"caml_process_pending_actions" from
 "<caml/signals.h>" executes any pending signal handlers and
-finalisers, Memprof callbacks, and requested minor and major garbage
-collections. In particular, it can raise asynchronous exceptions. It
-is recommended to call it regularly at safe points inside long-running
-non-blocking C code.
+finalisers, Memprof callbacks, thread preemption, and requested or
+incoming garbage collections events. In particular, it can raise
+asynchronous exceptions. It is recommended to call
+"caml_process_pending_actions" regularly at safe points inside
+long-running C code that holds the domain lock.
+
+In addition, in the rare case of long-running C code that holds the
+domain lock but does not allocate regularly, not calling
+"caml_process_pending_actions" regularly can cause all other domains
+to remain stuck while waiting for the C code to finish, due to pending
+garbage collection events.
 
 The variant \verb"caml_process_pending_actions_exn" is provided, that
-returns the exception instead of raising it directly into OCaml code.
-Its result must be tested using {\tt Is_exception_result}, and
-followed by {\tt Extract_exception} if appropriate. It is typically
-used for clean up before re-raising:
+does the same but returns the exception, instead of raising it
+directly into OCaml code. Its result must be tested using {\tt
+Is_exception_result}, and followed by {\tt Extract_exception} if
+appropriate. It is typically used for clean up before re-raising:
```

Comment:
I think this semantically neutral - personally, I'm not sure it's necessary (the function is referred to as a "variant" so it doesn't really need the addition of "does the same but" to me)

Reply:
This is stylistic, but having written the original sentence I can decide to change it.